hy/compiler.py

```diff
@@ -1719,7 +1736,18 @@ def hy_compile(tree, module_name, root=ast.Module, get_expr=False):
     if not get_expr:
         result += result.expr_as_stmt()
 
-    body = compiler.imports_as_stmts(tree) + result.stmts
+    body = []
+
+    # Pull out a single docstring and prepend to the resulting body.
+    if (len(result.stmts) > 0 and
+        issubclass(root, ast.Module) and
+        isinstance(result.stmts[0], ast.Expr) and
+        isinstance(result.stmts[0].value, ast.Str)):
+
+        body += [result.stmts.pop(0)]
+
+    body += sorted(compiler.imports_as_stmts(tree) + result.stmts,
+                   key=import_order_key)
```

Comment:
Couldn't you just write something like `key=lambda x: not (isinstance(x, ast.ImportFrom) and x.module == '__future__')`?

Reply:
Yeah, probably.  This started from an elaborate scheme to check `ast.ImportFrom` and multi-import `ast.Import`s, then I realized only `from __future__ import ...` statements needed to be handled and cut out parts from there.